tests/unit/mock_helpers.hpp

```diff
@@ -0,0 +1,82 @@
+  MOCK_METHOD(std::vector<msgs::CreateExpandResponse>, CreateExpand, (std::vector<msgs::NewExpand>));
+  MOCK_METHOD(std::vector<msgs::GetPropertiesResultRow>, GetProperties, (msgs::GetPropertiesRequest));
+  MOCK_METHOD(void, StartTransaction, ());
+  MOCK_METHOD(void, Commit, ());
+
+  MOCK_METHOD(storage::v3::EdgeTypeId, NameToEdgeType, (const std::string &), (const));
+  MOCK_METHOD(storage::v3::PropertyId, NameToProperty, (const std::string &), (const));
+  MOCK_METHOD(storage::v3::LabelId, NameToLabel, (const std::string &), (const));
+  MOCK_METHOD(storage::v3::LabelId, LabelToName, (const std::string &), (const));
+  MOCK_METHOD(const std::string &, PropertyToName, (storage::v3::PropertyId), (const));
+  MOCK_METHOD(const std::string &, LabelToName, (storage::v3::LabelId label), (const));
+  MOCK_METHOD(const std::string &, EdgeTypeToName, (storage::v3::EdgeTypeId type), (const));
+  MOCK_METHOD(std::optional<storage::v3::PropertyId>, MaybeNameToProperty, (const std::string &), (const));
+  MOCK_METHOD(std::optional<storage::v3::EdgeTypeId>, MaybeNameToEdgeType, (const std::string &), (const));
+  MOCK_METHOD(std::optional<storage::v3::LabelId>, MaybeNameToLabel, (const std::string &), (const));
+  MOCK_METHOD(bool, IsPrimaryLabel, (storage::v3::LabelId), (const));
+  MOCK_METHOD(bool, IsPrimaryKey, (storage::v3::LabelId, storage::v3::PropertyId), (const));
+};
+
+class MockedLogicalOperator : public plan::LogicalOperator {
+ public:
+  MOCK_METHOD(plan::UniqueCursorPtr, MakeCursor, (utils::MemoryResource *), (const));
+  MOCK_METHOD(std::vector<expr::Symbol>, ModifiedSymbols, (const expr::SymbolTable &), (const));
+  MOCK_METHOD(bool, HasSingleInput, (), (const));
+  MOCK_METHOD(std::shared_ptr<LogicalOperator>, input, (), (const));
+  MOCK_METHOD(void, set_input, (std::shared_ptr<LogicalOperator>));
+  MOCK_METHOD(std::unique_ptr<LogicalOperator>, Clone, (AstStorage * storage), (const));
+  MOCK_METHOD(bool, Accept, (plan::HierarchicalLogicalOperatorVisitor & visitor));
+};
+
+class MockedCursor : public plan::Cursor {
+ public:
+  MOCK_METHOD(bool, Pull, (Frame &, expr::ExecutionContext &));
+  MOCK_METHOD(void, PullMultiple, (MultiFrame &, expr::ExecutionContext &));
+  MOCK_METHOD(void, Reset, ());
+  MOCK_METHOD(void, Shutdown, ());
+};
+
+inline expr::ExecutionContext MakeContext(const expr::AstStorage &storage, const expr::SymbolTable &symbol_table,
+                                          RequestRouterInterface *router, IdAllocator *id_alloc) {
+  expr::ExecutionContext context;
+  context.symbol_table = symbol_table;
+  context.evaluation_context.properties = NamesToProperties(storage.properties_, router);
+  context.evaluation_context.labels = NamesToLabels(storage.labels_, router);
+  context.edge_ids_alloc = id_alloc;
+  context.request_router = router;
+  return context;
+}
+
+inline MockedLogicalOperator &BaseToMock(plan::LogicalOperator *op) {
+  return *static_cast<MockedLogicalOperator *>(op);
```

Comment:
Please use either `utils::DownCast` or `dynamic_cast`. Probaby `dynamic_cast<MockedLogicalOperator&>(*op) is the best as it will throw in case of invalid cast.

Reply:
I did consider `DownCast` but opted in for `static_cast` because I assumed that the name `BaseToMock` is enough to avoid accidental misuse (and UB). As there is no benefit of having the static_cast here other than a potential misuse I replaced it with dynamic_cast (to avoid the kType).